src/modules/disk_used.rs

```diff
@@ -0,0 +1,323 @@
+                    Some(name) => Some(Ok(name)),
+                    _ => None,
+                },
+                "usage" => Some(Ok(formatted_usage.as_str())),
+                "separator" if add_separator => Some(Ok(config.separator)),
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    match parsed {
+        Ok(parsed) => Ok(parsed),
+        Err(e) => Err(Box::new(e)),
+    }
+}
+
+fn should_display_disk(disk: &Disk, threshold: i64) -> bool {
+    let percentage = (disk.get_total_space() - disk.get_available_space()) as f64
+        / disk.get_total_space() as f64;
+
+    percentage >= (threshold as f64 * 0.01f64)
+}
+
+#[derive(Debug)]
+struct DiskNotFoundError {
+    path: PathBuf,
+}
+impl DiskNotFoundError {
+    fn new(path: PathBuf) -> Self {
+        DiskNotFoundError { path }
+    }
+}
+impl fmt::Display for DiskNotFoundError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "No disk found for path: {:?}!", self.path)
+    }
+}
+impl Error for DiskNotFoundError {}
+
+#[cfg(target_os = "windows")]
+#[derive(Debug)]
+struct WinAPIVolumeError {
+    desc: String,
+}
+#[cfg(target_os = "windows")]
+impl WinAPIVolumeError {
+    fn new(desc: String) -> Self {
+        WinAPIVolumeError { desc }
+    }
+}
+#[cfg(target_os = "windows")]
+impl fmt::Display for WinAPIVolumeError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.desc)
+    }
+}
+#[cfg(target_os = "windows")]
+impl Error for WinAPIVolumeError {}
+
+#[cfg(target_os = "windows")]
+fn get_drive_from_path<'a>(path: &PathBuf, disks: &'a [Disk]) -> Result<&'a Disk, Box<dyn Error>> {
+    use std::{ffi::OsString, iter::once, os::windows::prelude::*};
+    use winapi::um::{errhandlingapi::GetLastError, fileapi::GetVolumePathNameW};
+
+    let path: Vec<u16> = path.as_os_str().encode_wide().chain(once(0)).collect();
+    let mut volume_path_name: Vec<u16> = vec![0; 260];
+
+    let ret = unsafe {
+        GetVolumePathNameW(
+            path.as_ptr(),
+            volume_path_name.as_mut_ptr(),
+            volume_path_name.len() as u32,
+        )
+    };
+
+    if ret != 0 {
+        // Strip out the nulls from the end
+        while let Some(last) = volume_path_name.last() {
+            if *last == 0 {
+                volume_path_name.pop();
+            } else {
+                break;
+            }
+        }
+        let volume_path_name = OsString::from_wide(&volume_path_name);
+        log::info!("Got path name: {:?}", volume_path_name);
+        for disk in disks {
+            if volume_path_name == disk.get_mount_point() {
+                return Ok(disk);
+            }
+        }
+        Err(Box::new(DiskNotFoundError::new(PathBuf::from(
+            volume_path_name,
+        ))))
+    } else {
+        let last_error = unsafe { GetLastError() };
+        Err(Box::new(WinAPIVolumeError::new(format!(
```

Comment:
How about using `std::io::last_os_error()` instead of the custom error and logging the failure? Anyways the speed on windows appears to be unusable so it might make sense to just use sysinfo or not supporting windows at all.

Reply:
I didn't know that `std::io::last_os_error()` existed will switch to it. I will also look into if I can do anything about the performance

Reply:
I am going to leave the windows part in with the hopes that sysinfo gets faster on windows with future builds, since the module is disabled by default I don't think it should cause any issues.